esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -411,10 +411,99 @@ static void ensureMergeOrderedEsdlTransform(Owned<IPropertyTree> &dest, IPropert
     mergePTree(dest, copy);
 }
 
+void EsdlServiceImpl::configureBindingAccessMap(const IPTree* node, const char* service, const char* method, const char* serviceSecurity)
+{
+    StringBuffer methodSecurity;
+    StringArray  accessList;
+    StringBuffer accessName;
+    StringBuffer accessLevel;
+    Owned<MethodAccessMap> methodAccess;
+
+    if (!isEmptyString(serviceSecurity))
+    {
+        // Service and method access requirements are combined by default, but the method may opt
+        // to ignore the service requirements.
+        //
+        // Consider a service level requirement of "{$method}Access" that applies a method-specific
+        // requirement to every method. Some methods, such as Ping or EchoTest, may require no
+        // security. The 'inherited' requirement may be incorrect for other methods, perhaps because
+        // the name is too long. Methods must be able to override the service's settings.
+        if (!node->getPropBool("@override_service_auth_feature", false))
+            methodSecurity.set(serviceSecurity).append(',');
+    }
+    methodSecurity.append(node->queryProp("@auth_feature"));
+    methodSecurity.replaceString("{$service}", service);
+    methodSecurity.replaceString("{$method}", method);
+
+    accessList.appendList(methodSecurity, ",");
+    if (accessList.ordinality() == 0)
+    {
+        // Either the binding's service and method both omit the auth_feature attribute or service
+        // level security is given and the method explicitly overrides it without specifying its
+        // own entries. The binding defers.
+        return;
+    }
+
+    // At least one security entry exists in the binding. The binding will define a method access
+    // map (even if empty) so it can take precedence.
+
+    methodAccess.setown(new MethodAccessMap());
+    m_bindingAccessMaps.setValue(method, methodAccess.getLink());
+    ForEachItemIn(idx, accessList)
+    {
+        auto accessEntry = accessList.item(idx);
+        auto entryLen = strlen(accessEntry);
```

Comment:
In this case, strlen is unlikely to be replaced with something that returns a type other than size_t, so this may not be the best example of another good use case for auto (and the following use of decltype). This other use case is when the code is not dependent on a specific type. Is it const or not const, signed or unsigned, 32-bit or 64-bit, Auto provides an appropriate type, and decltype ensures a compatible value that should never generate a compiler warning.

I'll change (both).